src/node.cc

```diff
@@ -3803,6 +3810,31 @@ static void ParseArgs(int* argc,
       }
       args_consumed += 1;
       config_userland_loader = module;
+    } else if (strncmp(arg, "--mode=", 7) == 0) {
+      if (!config_experimental_modules) {
+        fprintf(stderr, "%s: %s requires --experimental-modules be enabled\n",
+            argv[0], arg);
+        exit(9);
+      }
+      config_main_mode = arg + 7;
+      if (config_main_mode != "esm") {
+        fprintf(stderr, "%s: \"%s\" is not a valid module mode\n", argv[0],
```

Comment:
I'm not a pro in c++ but is `config_main_mode = arg + 7` a number result? If so would it never match `"esm"` in the `if` statement.

Reply:
`config_main_mode` is a string. I assume `arg + 7` is the pointer to the null-terminated string starting 7 bytes from the pointer `arg`, skipping the `--mode=` (7 bytes). But it's not the *most* straight-forward to follow.